docs/cow-protocol/tutorials/arbitrate/README.mdx

```diff
@@ -133,3 +133,125 @@ sequenceDiagram
     Note over autopilot: End
 ```
 
+## Run locally end to end
+
+In this tutorial we will run all the components locally on Ethereum mainnet, including the orderbook, autopilot, driver and solver engine. All these components have their own configuration parameters. For the sake of simplicity, we will use the bare minimum configuration for all of them.
+
+## Solver engine configuration file
+
+baseline.toml
+```
+chain-id = "1" # Ethereum mainnet
+base-tokens = [
+    "0xc02aaa39b223fe8d0a0e5c4f27ead9083c756cc2",
+    "0x6B175474E89094C44Da98b954EedeAC495271d0F",
+    "0xA0b86991c6218b36c1d19D4a2e9Eb0cE3606eB48",
+    "0xdAC17F958D2ee523a2206206994597C13D831ec7",
+    "0xc00e94Cb662C3520282E6f5717214004A7f26888",
+    "0x9f8F72aA9304c8B593d555F12eF6589cC3A579A2",
+    "0x2260FAC5E5542a773Aa44fBCfeDf7C193bc2C599",
+    "0x6810e776880C02933D47DB1b9fc05908e5386b96",
+]
+max-hops = 2
+max-partial-attempts = 5
+risk-parameters = [0,0,0,0]
+```
+
+## Driver configuration file
+
+driver.toml
+```
+[[solver]]
+name = "baseline" # Arbitrary name given to this solver, must be unique
+endpoint = "http://0.0.0.0:7872"
+absolute-slippage = "40000000000000000" # Denominated in wei, optional
+relative-slippage = "0.1" # Percentage in the [0, 1] range
+account = "0x01246d541E732D7F15d164331711edfF217e4665"
+
+[submission]
+gas-price-cap = 1e12
+additional-tip-percentage = 0.05
+
+[[submission.mempool]]
+mempool = "public"
+
+[liquidity]
+base-tokens = [
+    "0xc02aaa39b223fe8d0a0e5c4f27ead9083c756cc2",
+    "0x6B175474E89094C44Da98b954EedeAC495271d0F",
+    "0xA0b86991c6218b36c1d19D4a2e9Eb0cE3606eB48",
+    "0xdAC17F958D2ee523a2206206994597C13D831ec7",
+    "0xc00e94Cb662C3520282E6f5717214004A7f26888",
+    "0x9f8F72aA9304c8B593d555F12eF6589cC3A579A2",
+    "0x2260FAC5E5542a773Aa44fBCfeDf7C193bc2C599",
+    "0x6810e776880C02933D47DB1b9fc05908e5386b96",
+]
+
+[[liquidity.uniswap-v2]] # Uniswap V2 configuration
+preset = "uniswap-v2"
+```
+
+## Run script
+
+run.sh
+```bash
+#!/bin/bash
+export REPO_DIR='<path to repo>'; # checked out from https://github.com/cowprotocol/services
+export ETHRPC='<eth rpc url>'; # e.g. https://mainnet.infura.io/v3/<your infura key>
+export TOML_TRACE_ERROR="1";
+
+export NODE_URL=$ETHRPC;
+export SIMULATION_NODE_URL=$ETHRPC;
+export ETH_RPC=$ETHRPC;
+export TRACING_NODE_URL=$ETHRPC;
+
+# Fetch current auction from public orderbook and run auction locally
+# If you want to run an orderbook locally comment out this line
+# export SHADOW='https://api.cow.fi/mainnet';
+
+export SETTLE_INTERVAL=5
+export GAS_ESTIMATORS='Native,Web3';
+export PRICE_ESTIMATORS='None';
+export BLOCK_STREAM_POLL_INTERVAL_SECONDS='1'
+export TOKEN_OWNER_FINDERS='liquidity,blockscout,ethplorer,solvers';
+export SOLVER_TOKEN_OWNERS_CACHE_UPDATE_INTERVALS="300"
```

Comment:
We can drop the token owners cache thing.
I only had it in my example because the yearn team needed some help setting this up and they relied on this feature.

Reply:
Removed.